File: Strings/LengthOfLastWord.py
# LeetCode 58 - Length of Last Word
# Time Complexity: O(n) | Space Complexity: O(1)


class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        end = len(s) - 1

        while end >= 0 and s[end] == " ":
            end -= 1

        start = end
        while start >= 0 and s[start] != " ":
            start -= 1

        return end - start

File: Strings/LengthOfLastWord.go
// LeetCode 58 - Length of Last Word
// Time Complexity: O(n) | Space Complexity: O(1)
package main

func lengthOfLastWord(s string) int {
	end := len(s) - 1

	for end >= 0 && s[end] == ' ' {
		end--
	}

	start := end
	for start >= 0 && s[start] != ' ' {
		start--
	}

	return end - start
}
